app/src/main/java/net/squanchy/schedule/ScheduleService.kt

```diff
@@ -1,74 +1,92 @@
 import net.squanchy.schedule.domain.view.Schedule
 import net.squanchy.schedule.domain.view.SchedulePage
+import net.squanchy.schedule.domain.view.Track
 import net.squanchy.service.firebase.FirebaseAuthService
-import net.squanchy.service.firebase.FirebaseDbService
-import net.squanchy.service.firebase.model.FirebaseDay
-import net.squanchy.service.firebase.model.FirebaseDays
-import net.squanchy.service.repository.EventRepository
+import net.squanchy.service.firestore.FirestoreDbService
+import net.squanchy.speaker.domain.view.Speaker
+import net.squanchy.support.lang.Checksum
+import net.squanchy.support.lang.optional
+import org.joda.time.DateTimeZone
+import org.joda.time.LocalDate
 import org.joda.time.LocalDateTime
-import org.joda.time.format.DateTimeFormat
 
-class ScheduleService internal constructor(
-        private val dbService: FirebaseDbService,
-        private val authService: FirebaseAuthService,
-        private val eventRepository: EventRepository
-) {
-
-    fun schedule(onlyFavorites: Boolean): Observable<Schedule> {
-        return authService.ifUserSignedInThenObservableFrom { userId ->
-            val daysObservable = dbService.days()
-
-            val eventsObservable = eventRepository.events(userId)
-                    .map { events -> if (onlyFavorites) events.filter { it.favorited } else events }
-                    .map { it.groupBy { it.dayId } }
-
-            eventsObservable
-                    .withLatestFrom(daysObservable, combineEventsById())
-                    .subscribeOn(Schedulers.io())
-        }
-    }
-
-    private fun combineEventsById(): BiFunction<Map<String, List<Event>>, FirebaseDays, Schedule> {
-        return BiFunction { eventsMap, (days) ->
-            val pages = eventsMap.keys
-                    .map(findDayById(days!!))
-                    .filterNotNull()
-                    .map(toSchedulePage(eventsMap))
-                    .sortedBy { it.date }
-
-            Schedule.create(pages)
-        }
-    }
+interface ScheduleService {
+    fun schedule(onlyFavorites: Boolean): Observable<Schedule>
+    fun currentUserIsSignedIn(): Observable<Boolean>
+}
 
-    private fun findDayById(days: List<FirebaseDay>): (String) -> FirebaseDay? {
-        return { dayId ->
-            days.find { (id) ->
-                id == dayId
+class FirestoreScheduleService(
+        private val authService: FirebaseAuthService,
+        private val dbService: FirestoreDbService,
+        private val checksum: Checksum
+) : ScheduleService {
+    override fun schedule(onlyFavorites: Boolean): Observable<Schedule> {
+        return dbService.scheduleView()
+            .map {
+                it.map { schedulePage ->
+                    SchedulePage(
+                            schedulePage.day.id,
+                            LocalDate(schedulePage.day.date),
+                            schedulePage.events.map {
+                                Event(
```

Comment:
This is getting a bit complicated. Can we extract some mapping functions for event, track and speakers? 
Something like:
`(FirestoreSchedulePage) -> SchedulePage `
`(FirestoreEvent) -> Event` 
and so on

Reply:
I would leave this for the next PR